pkg/cmd/apply.go

```diff
@@ -317,7 +341,17 @@ func runApplyUpgradeIfNeeded(s *state.State, opts *applyOpts) error {
 
 	tasksToRun := tasks.WithResources(nil)
 	if upgradeNeeded || opts.ForceUpgrade {
-		tasksToRun = tasks.WithUpgrade(nil)
+		// disable case, we do this as early as possible.
+		if s.ShouldDisableEncryption() {
+			tasksToRun = tasks.WithDisableEncryptionProviders(tasksToRun, s.LiveCluster.EncryptionConfiguration.Custom)
+		}
+
+		tasksToRun = tasks.WithUpgrade(tasksToRun)
```

Comment:
With recent changes (automate certificate renew) we got `WithResources()` included into `WithUpgrade()`. So now  stacking them here is kinda duplication.